boards: arm: Add support for the VBLUno51 board

iotmember opened the pull request:
+ VBLUno51 board
    nRF51822
    Bluetooth Low Energy
    DAPLink interface
    UNO pinout
    4 power

+ Wiki: https://vngiotlab.github.io/vbluno/

+ The following samples were tested:
    hello_world
    basic/button
    basic/blinky
    bluetooth/peripheral_hr
    bluetooth/beacon

Signed-off-by: Nam Do <robotden@gmail.com>

iotvietmember commented:
@galak  @cvinayak : I closed the old PR and create new PR.
I am really sorry about this inconvenience.

galak commented on boards/arm/nrf51_vbluno51/Makefile.board:
```diff
@@ -0,0 +1,8 @@
+
+PYOCD_TARGET = nrf51
+
+NRF_FAMILY = NRF51
```
if you are using PYOCD as the flash script why do we need NRF_FAMILY set or exported?

galak commented on dts/arm/nrf51_vbluno51.dts:
```diff
@@ -0,0 +1,25 @@
+
+/ {
+	model = "VNG VBLUno51 BLE board";
+	compatible = "nordic,pca10028-dk", "nordic,nrf51822-qfac",
```
This should be:

compatible = "vng,vbluno51", "nordic,nrf51822-qfac", "nordic,nrf51822";

(most specific to least)

galak reviewed (changes requested):
A few minor things

carlescufi commented:
@iotvietmember you should prefix your commit with "boards:". Something like:

boards: Add support for the VBLUno51 board

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,156 @@
+********
+
+Zephyr applications use the nrf51_vbluno51 board configuration to run on the
+VBLUno51 board - VNG Bluetooth Low Energy UNO nRF51822 hardware. It provides support for the Nordic Semiconductor
```
should keep line length < 80 chars if possible

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,156 @@
+Flashing
+========
+
+The VBLUno51 board have on-board DAPLink (CMSIS-DAP) interface for flashing and debugging. You do not need any other programming device.
```
another long line

dbkinder replied:
change "have" to "has"

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,156 @@
+You only need to install pyOCD tool (https://pypi.python.org/pypi/pyOCD)
+
+This tutorial uses the sample application
+blinky :file:`$ZEPHYR_BASE/samples/basic/blinky`.
```
recommend instead of pointing to the file, use a reference to the blinky sample doc:
    :ref:`blinky-sample`

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,156 @@
+the board are working properly with Zephyr:
+
+.. code-block:: console
+   samples/hello_world
```
needs a blank line after the `.. code-block:: console` line and before the first line of the code block.  That said, this is really just a list of samples, so I'd recommend a rewrite as just a list of references to the sample docs:

   * :ref:`hello_world`
   * :ref:`blinky-sample`
   * :ref:`button-sample`
   * :ref:`bluetooth-beacon-sample`
   * :ref:`peripheral_hr`

dbkinder reviewed (changes requested):
a few recommended changes

iotvietmember commented:
@galak, @dbkinder, @carlescufi : I have just fixed the recommended problems. Many thanks

galak commented:
Nit picking here, but commit message should be (mostly picking on the space after ':'):

board: arm: Add support for the VBLUno51 board

galak commented:
@dbkinder look good?

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,138 @@
+
+This tutorial uses the blinky application :ref:`blinky-sample`
+
+See getting started page: https://www.zephyrproject.org/doc/getting_started/getting_started.html
```
Better would be

     See the :ref:`getting_started` for general information on setting up 
     your development environment.

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,138 @@
+
+See getting started page: https://www.zephyrproject.org/doc/getting_started/getting_started.html
+
+#. To build the Zephyr kernel, enter:
```
#. To build the Zephyr kernel along with the blinky sample app, enter:

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,138 @@
+      $ cd $ZEPHYR_BASE/samples/basic/blinky
+      $ make BOARD=nrf51_vbluno51
+
+#. To build and flash hex file to board
```
#. To build and flash the generated binary image to the board, enter:

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,138 @@
+      $ make BOARD=nrf51_vbluno51 flash
+
+Testing the VBLUno51 with Zephyr: buttons, LEDs, UART, BLE
+********************************************
```
line of asterisks needs to be same length as the title

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,138 @@
+Testing the VBLUno51 with Zephyr: buttons, LEDs, UART, BLE
+********************************************
+
+.. code-block:: console
```
Sorry my previous comment wasn't clear. Don't use a code-block for this list.

     Here are some sample applications that you can use to test different
     components on the VBLUno51 board:

     * :ref:`hello_world`
     * :ref:`blinky-sample`
     * :ref:`button-sample`  
     * :ref:`bluetooth-beacon-sample`
     * :ref:`peripheral_hr`

dbkinder reviewed (changes requested):
Almost.  A few more suggested edits.

iotvietmember commented:
Hi @dbkinder, I have just fixed the recommended problems. Many thanks for your help.

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,140 @@
+
+.. figure:: img/vbluno51_frizting.png
+     :align: center
+     :alt: nRF51_VBLUno51 Frizting part
```
The correct spelling is "Fritzing"

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,140 @@
+Overview
+********
+
+Zephyr uses the nrf51_vbluno51 board configuration to run on the VBLUno51 board.
```
I think this should read: 

    Zephyr uses the nrf51_vbluno51 board configuration to run on the VBLUno51 board,
    a VNG Bluetooth Low Energy UNO using an nRF51822 ARM processor.

dbkinder reviewed (changes requested):
sorry, missed a couple issues

iotmember committed:
boards: arm: Add support for the VBLUno51 board

    + VBLUno51 board
        nRF51822
        Bluetooth Low Energy
        DAPLink interface
        UNO pinout
        4 power

    + Wiki: https://vngiotlab.github.io/vbluno/

    + The following samples were tested:
        hello_world
        basic/button
        basic/blinky
        bluetooth/peripheral_hr
        bluetooth/beacon

Signed-off-by: Nam Do <robotden@gmail.com>

iotvietmember commented:
@dbkinder, thanks for your help

iotmember committed:
Merge branch 'arm' into add_vbluno51_board

dbkinder commented on boards/arm/nrf51_vbluno51/doc/nrf51_vbluno51.rst:
```diff
@@ -0,0 +1,140 @@
+
+.. target-notes::
+
+.. _VBLUno51 website: http://iotviet.com.vn/store/detail?id=2
```
I didn't see an obvious way to construct a link so this page opens in English, but there's a button in the top to switch to English from the default Vietnamese (and the Wiki is in English), so OK as is.

dbkinder reviewed (approved):
+1 thanks!

iotvietmember commented:
Many thanks
